### ISSUE TITLE
[Accessibility]: Branches page: Undescriptive label on branch list

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35461]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34424

### Problem description

Clicking a branch in the list will **load the current repository using that branch as the selected revision**. This is only obvious when a visual user can see the current revision has updated - not obvious for non-visual users.

Image for reference:
<img width="769" alt="image" src="https://user-images.githubusercontent.com/9516420/168571235-75a81d62-1c92-4da9-897c-69ffbd8dee27.png">

### Expected behavior

Add a descriptive label to the links that only screen readers can pick up. For example:

`aria-label="View this repository using this branch as the selected revision"`

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
